middleware.go

```diff
@@ -0,0 +1,23 @@
+package jwt
+
+import "net/http"
+
+func RequireClaim(handler http.Handler, idpKey interface{}, header, claimKey, expectedClaimValue string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, claims, err := GetClaimsFromRequestWithValidation(r, idpKey)
```

Comment:
Seems like this shouldn't validate the claims here, validation should happen in a separate middleware

Reply:
A jwt is a signed json blob, so why shouldn't we validate the signature and expiry in the auth middleware, especially when it's just a 10 line middleware?

Reply:
I refactored that so that we now extract and validate the token in a seperate middleware and store the claims in the requests context. With that in place it will be easy to continue working on the claims in http handlers further down the handler tree.

@LucasRoesler Ping :) This can be reviewed again.